### PR TITLE
Rework and fix "disable_non_core_proposals"

### PR DESCRIPTION
**Pull request by rhafer**
Relying on chef search to find relevant nodes which have the element roles
assinged doesn't work as the nodes are in "crowbar_upgrade" state, where they
the lack the element roles.
This implementation also reduces the amount of node.save calls to one per node.

### CODE REVIEW

File: crowbar_framework/app/models/crowbar_service.rb [hunk 1]
```diff
@@ -288,21 +288,20 @@ def disable_non_core_proposals
...
+      active_non_core_roles.each do |role|
+        roles_to_remove = role.elements.keys
+        roles_to_remove << role.name
+        roles_to_remove.each do |delete_role|
           node.delete_from_run_list(delete_role)
-          node.save
         end
       end
-      role.destroy
+      node.save
```
Comment (aplanas) [inline crowbar_framework/app/models/crowbar_service.rb]: As I understand in the original code, the role is also removed, but not in the new version. Is this change OK?

**Comment by jsuchome** [comment]
LGTM

File: crowbar_framework/app/models/crowbar_service.rb (hunk 1 above)
Comment (MaximilianMeister) [inline crowbar_framework/app/models/crowbar_service.rb]: happens below in 304

**Comment by MaximilianMeister** [comment]
maybe re-add some logging, but codewise +1

**Comment by vuntz** [comment]
+1

**Comment by aplanas** [comment]
+1

**Pull request merged**